comp/core/telemetry/telemetry.go

```diff
@@ -54,14 +56,16 @@ func newTelemetry() Component {
 
 // Same as `newTelemetry“ without the global.
 // Can be merged with `newTelemetry` when the global is removed
-func newMock() Component {
+func newMock() (Component, Mock) {
```

Comment:
As an alternative to return 2 values, you can use [fx.As](https://pkg.go.dev/go.uber.org/fx#As). Feel free keep you code, if you think it is more obvious to you.